process/block/metablock.go

```diff
@@ -205,6 +211,11 @@ func (mp *metaProcessor) ProcessBlock(
 	return nil
 }
 
+// SetConsensusRewardAddresses - sets the reward addresses for the current consensus group
+func (mp *metaProcessor) SetConsensusRewardAddresses(consensusRewardAddresses []string) {
```

Comment:
Why implement this at all? We could provide access to `specialAddressHandler` where needed and call the `SetConsensusRewardAddresses` directly on it.

Reply:
We anyway need it in processor, and consensus has access to processor.